src/main/java/edu/wpi/grip/ui/preview/PreviewsView.java

```diff
@@ -28,12 +33,14 @@
 
     private final EventBus eventBus;
     private final List<OutputSocket<?>> previewedSockets;
+    private PipelineView pipeline;//This is used to determine the order the previews are shown in.
 
-    public PreviewsView(EventBus eventBus) {
+    public PreviewsView(EventBus eventBus, PipelineView pipeline) {
         checkNotNull(eventBus);
 
         this.eventBus = eventBus;
         this.previewedSockets = new ArrayList<>();
+        this.pipeline = pipeline;
```

Comment:
Can you make this `this.pipeline checkNotNull(pipeline, "The Pipeline can not be null");